packages/firestore/src/core/sync_engine.ts

```diff
@@ -231,14 +239,26 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
           );
           this.queryViewsByQuery.set(query, data);
           this.queryViewsByTarget[queryData.targetId] = data;
-          this.viewHandler!([viewChange.snapshot!]);
-          if (this.isPrimary) {
-            this.remoteStore.listen(queryData);
-          }
+          return viewChange.snapshot!;
         });
     });
   }
 
+  /**
+   * Reconcile the list of synced documents in the local views with those from
+   * persistence.
+   */
+  // PORTING NOTE: Multi-tab only.
+  private async synchronizeLocalView(queryData: QueryData): Promise<void> {
```

Comment:
This method seems to only need the targetId, not the full queryData.

Reply:
Good catch. Fixed.